test/benchmark/remote/run.sh

```diff
@@ -0,0 +1,158 @@
+
+# change to script directory
+cd "${0%/*}" || exit
+
+function main() {
+  while [[ "$#" -gt 0 ]]; do
+      case $1 in
+          --all) run_all; exit 0 ;;
+          --create_machine) create_machine; exit 0 ;;
+          --clone_repository) clone_repository; exit 0;;
+          --delete_machine) delete_machine; exit 0;;
+          --install_dependencies) install_dependencies; exit 0;;
+          --benchmark) benchmark; exit 0;;
+          --prepare) prepare; exit 0;;
+          --checkout) checkout "$2" ; exit 0;;
+          --ssh) interactive_ssh; exit 0;;
+          *) echo "Unknown parameter passed: $1"; exit 1 ;;
+      esac
+      shift
+  done
+
+  print_help
+}
+
+function print_help() {
+  echo "Valid arguments include:"
+  echo ""
+  echo "  --all               Run everything, including machine creation & destruction"
+  echo "  --prepare           Create Machine & run all the steps prior to benchmark execution"
+  echo "  --create_machine    Only create machine"
+  echo "  --delete_machine    Stop & Delete running machine"
+  echo "  --clone_repository  Clone and checkout Weaviate repo at specified commit"
+  echo "  --checkout          Checkout arbitrary branch or commit"
+  echo "  --ssh               Interactive SSH session"
+}
+
+function run_all() {
+  trap delete_machine EXIT
+
+  check
+  prepare
+  benchmark
+}
+
+function prepare() {
+  check
+  create_machine
+  install_dependencies
+  clone_repository
+}
+
+function check() {
+  echo_green "Checking required dependencies"
+  if ! command -v gcloud &> /dev/null
+  then
+      echo_red "Missing gcloud binary"
+      return 1
+  fi
+
+  if ! command -v terraform &> /dev/null
+  then
+      echo_red "Missing terraform binary"
+      return 1
+  fi
+
+  echo "Ready to go!"
+}
+
+function create_machine() {
+  (cd terraform && terraform init)
+  (cd terraform && terraform apply -auto-approve)
+  echo "Sleeping for 10s, so first ssh doesn't fail. This should be improved through polling"
+  sleep 10
+}
+
+function delete_machine() {
+  (cd terraform && terraform destroy -auto-approve)
+}
+
+function install_dependencies() {
+  ssh_command "sudo apt-get update && sudo apt-get install -y git git-lfs curl"
+  ssh_command "ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts"
+  install_go
+  install_docker
+}
+
+function install_go {
+  ssh_command "curl -Lo go.tar.gz $GOVERSION"
+  ssh_command "sudo rm -rf /usr/local/go && sudo tar -C /usr/local -xzf go.tar.gz"
+  ssh_command 'echo '"'"'PATH=$PATH:/usr/local/go/bin'"'"' >> ~/.profile'
+  ssh_command "go version"
+}
+
+function install_docker() {
+  ssh_command "if ! command -v docker &> /dev/null; then curl -fsSL https://get.docker.com -o get-docker.sh && sh ./get-docker.sh; fi"
+  ssh_command "sudo groupadd docker || true"
+  ssh_command 'sudo usermod -aG docker $USER'
+}
+
+function clone_repository() {
+  ssh_command "cd; [ ! -d weaviate ] && git clone https://github.com/semi-technologies/weaviate.git weaviate || true"
```

Comment:
I think it should be possible to clone the branch we want directly using `git clone --depth 1 --branch <branchname> <remote-repo-url>`

Reply:
Good point, also the shallow clone should probably save some time. I'll adapt it.

Reply:
Updated, it now also checks out the branch instead of the commit, so the repo is never in a detached head state. I don't think it should matter much, since the remote machine is very unlikely to ever commit something. But still, it feels clear, thanks for the suggestion.